compiler/rustc_borrowck/src/borrowck_errors.rs

```diff
@@ -233,17 +243,11 @@ impl<'cx, 'tcx> crate::MirBorrowckCtxt<'cx, 'tcx> {
         borrow_span: Span,
         desc: &str,
     ) -> DiagnosticBuilder<'cx, ErrorGuaranteed> {
-        let mut err = struct_span_err!(
-            self,
-            span,
-            E0506,
-            "cannot assign to {} because it is borrowed",
-            desc,
-        );
-
-        err.span_label(borrow_span, format!("borrow of {} occurs here", desc));
-        err.span_label(span, format!("assignment to borrowed {} occurs here", desc));
-        err
+        let err = AssignBorrowErr { desc, span };
```

Comment:
Does `desc` contain English words? If so, we might need to modify that because it otherwise would not be translatable.

Reply:
`desc` is called [here](https://github.com/rust-lang/rust/blob/master/compiler/rustc_borrowck/src/diagnostics/mod.rs), I guess it contains a English word :]


```rust
    pub(super) fn describe_any_place(&self, place_ref: PlaceRef<'tcx>) -> String {
        match self.describe_place(place_ref) {
            Some(mut descr) => {
                // Surround descr with `backticks`.
                descr.reserve(2);
                descr.insert(0, '`');
                descr.push('`');
                descr
            }
            None => "value".to_string(),
        }
    }
```

Reply:
I assume `describe_place` also contains English words though - some are fine, things like "enum variant" should still be able to be translated by matching on `{$desc}` but other words might make it difficult.

Reply:
I think I'm [pretty sure](https://github.com/rust-lang/rust/blob/36e530cb08950f1d03ab733e43ecec2802d099cf/compiler/rustc_borrowck/src/diagnostics/mod.rs#L185) "value" is the only english word. 

the `String` is stored in `buf`:

```rust
// pushed a symbol
let mut ok = self.append_local_to_string(local, &mut buf);

// pushed a symbol
buf.push_str(self.infcx.tcx.item_name(*def_id).as_str());

// my rust-analyzer don't recognize `place` here but it's from
// /rust/compiler/rustc_middle/src/ty/closure.rs:335
// which returns a symbol if not `bug!`
buf = self.upvars[var_index].place.to_string(self.infcx.tcx);

// pushed a symbol
buf.push_str(&field_name_str);
```